src/aind_metadata_upgrader/data_description_upgrade.py

```diff
@@ -56,11 +62,14 @@ def upgrade_modality(cls, old_modality: Union[str, dict, Modality, None]) -> Opt
 
         """
         if type(old_modality) is str and cls.legacy_name_mapping.get(old_modality.lower()) is not None:
```

Comment:
this line is already checking that the old_modality exists in the lookup table. you don't need to check again down below.